### PR TITLE
Higher order basis funs

### PR DESCRIPTION
**Pull request by jayprakashdav**
I found the problem. It was indeed in the permutation of vertices. The permutation vector contains the indices of the vertex numbering, not the vertex numbering itself. That is only different for 2 combinations i.e. {2,3,1} => [3,1,2] and {3,1,2} => [2,3,1]. The rest of the four permutations are identical maps, i.e. {1,2,3} => [1,2,3]; {2,1,3} => [2,1,3]; {1,3,2} => [1,3,2]; {3,2,1} => [3,2,1]

### CODE REVIEW

**Comment by krcools** [comment]
This looks great and yes I agree now that the inverse Jacobian determinant should be part of the  push forward formula...

File: src/bases/rt2space.jl [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    cps = cellpairs(mesh, edges, dropjunctionpair=true)
+    ids = findall(x -> x>0, cps[2,:])
+    raviartthomas2(mesh, cps[:,ids])
+end
+
+divergence(X::RT2Basis, geo, fns) = LagrangeBasis{0,-1,1}(geo, fns, deepcopy(positions(X)))
+ntrace(X::RT2Basis, geo, fns) = LagrangeBasis{0,-1,1}(geo, fns, deepcopy(positions(X)))
+
+
+function LinearAlgebra.cross(::NormalVector, s::RT2Basis)
```
Comment (krcools) [inline src/bases/rt2space.jl:126]: This does not make sense I think for the RT2 basis. Its rotated version should be the Nd2 space, not the Nd(1) space.

Comment (jayprakashdav) [reply src/bases/rt2space.jl:126]: Yes. I am commenting this part for now but will revise it when I push the ND2Basis.

**Comment by krcools** [comment]
Thanks for this, looks good. Can I just ask to be conservative with the exports? No way back once we export. So for a release or two its best to try out this functionality with fully qualified calls.

File: test/test_rt2.jl [hunk 2]
```diff
@@ -0,0 +1,23 @@
+using CompScienceMeshes
+using BEAST
+
+using Test
+
+"""
```
Comment (krcools) [inline test/test_rt2.jl:6]: This should be a comment, not a docstring. It is currently causing the test suite to fail.

File: test/test_nd2.jl [hunk 3]
```diff
@@ -0,0 +1,24 @@
+using CompScienceMeshes
+using BEAST
+
+using Test
+
+
+"""
```
Comment (krcools) [inline test/test_nd2.jl:7]: Comments instead of docstring here as well

**Pull request merged**